src/prof_data.c

```diff
@@ -1492,13 +1491,11 @@ prof_tctx_destroy(tsd_t *tsd, prof_tctx_t *tctx) {
 	}
 	malloc_mutex_unlock(tsd_tsdn(tsd), gctx->lock);
 	if (destroy_gctx) {
-		prof_gctx_try_destroy(tsd, prof_tdata_get(tsd, false), gctx,
-		    tdata);
+		prof_gctx_try_destroy(tsd, prof_tdata_get(tsd, false), gctx);
 	}
 
-	malloc_mutex_assert_not_owner(tsd_tsdn(tsd), tctx->tdata->lock);
-
 	if (destroy_tdata) {
+		malloc_mutex_assert_not_owner(tsd_tsdn(tsd), tdata->lock);
```

Comment:
Good catch! Let’s refactor this a bit, to destroy tdata right after unlock? I’d say even make tdata a scoped (I.e. make line 1440-1450 scoped) variable and comment it cannot be accessed outside of the scope.

Reply:
I don't think the comment is necessary after making `tdata` a scoped variable?

Reply:
Feel free to not do the comment, if the code is self explanatory enough.